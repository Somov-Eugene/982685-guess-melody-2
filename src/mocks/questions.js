const questions = [
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/f/fc/Russian_Anthem_chorus.ogg`,
        genre: `rock`,
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/f/fc/Russian_Anthem_chorus.ogg`,
        genre: `pop`,
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/f/fc/Russian_Anthem_chorus.ogg`,
        genre: `jazz`,
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/f/fc/Russian_Anthem_chorus.ogg`,
        genre: `rock`,
      },
    ],
  },

  {
    type: `artist`,
    song: {
      artist: `Jim Beam`,
      src: `https://upload.wikimedia.org/wikipedia/commons/f/fc/Russian_Anthem_chorus.ogg`,
    },
    answers: [
      {
        picture: `http://placehold.it/134x134`,
        artist: `John Snow`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jim Beam`,
      },
    ],
  }
];

export default questions;
