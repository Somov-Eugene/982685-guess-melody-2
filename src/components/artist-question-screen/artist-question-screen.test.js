import React from 'react';
import renderer from 'react-test-renderer';

import questions from '../../mocks/questions';
import createNodeMock from '../../mocks/test-utility';
import ArtistQuestionScreen from './artist-question-screen';

describe(`Проверка <ArtistQuestionScreen>`, () => {
  it(`компонент <ArtistQuestionScreen> корректно отрисован`, () => {
    const question = 1;
    const currentQuestion = questions[question];
    const handleAnswer = jest.fn();
    const options = {createNodeMock};

    const tree = renderer
      .create(<ArtistQuestionScreen
        question = {currentQuestion}
        onAnswer = {handleAnswer}
      />, options)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
