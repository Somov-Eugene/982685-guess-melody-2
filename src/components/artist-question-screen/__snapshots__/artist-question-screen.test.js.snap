// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка <ArtistQuestionScreen> компонент <ArtistQuestionScreen> корректно отрисован 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <button
      className="track__button track__button--play"
      disabled={true}
      onClick={[Function]}
      type="button"
    />
    <div
      className="track__status"
    >
      <audio />
    </div>
  </div>
  <form
    className="game__artist"
    onChange={[MockFunction]}
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-0"
        name="answer"
        type="radio"
        value="John Snow"
      />
      <label
        className="artist__name"
        htmlFor="artist-0"
      >
        <img
          alt="John Snow"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        John Snow
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-1"
        name="answer"
        type="radio"
        value="Jack Daniels"
      />
      <label
        className="artist__name"
        htmlFor="artist-1"
      >
        <img
          alt="Jack Daniels"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Jack Daniels
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-2"
        name="answer"
        type="radio"
        value="Jim Beam"
      />
      <label
        className="artist__name"
        htmlFor="artist-2"
      >
        <img
          alt="Jim Beam"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Jim Beam
      </label>
    </div>
  </form>
</section>
`;
